networkit/cpp/centrality/GroupClosenessLocalSearch.cpp

```diff
@@ -0,0 +1,1168 @@
+                checkDistance[u] = d;
+            } else if (d < checkDistance2[u])
+                checkDistance2[u] = d;
+        });
+    }
+
+    const auto &curDistance = parallel ? distanceGlobal[omp_get_thread_num()] : distance;
+    const auto &curDistance2 = parallel ? distance2Global[omp_get_thread_num()] : distance2;
+
+    G->forNodes([&](node u) {
+        if (group.find(u) != group.end())
+            return;
+        assert(curDistance2[u] == checkDistance2[u]);
+        assert(curDistance[u] == checkDistance[u]);
+    });
+}
+
+template <>
+void GroupClosenessLocalSearchImpl<edgeweight>::runSanityCheck(node v, bool parallel) const {
+    auto tmpGroup = group;
+    if (v != none)
+        tmpGroup.insert(v);
+
+    const count n = G->upperNodeIdBound();
+    std::vector<edgeweight> checkDistance(n, std::numeric_limits<edgeweight>::max());
+    std::vector<edgeweight> checkDistance2(n, std::numeric_limits<edgeweight>::max());
+    for (const auto source : tmpGroup) {
+        Traversal::DijkstraFrom(*G, source, [&](node u, edgeweight d) {
+            if (d < checkDistance[u]) {
+                checkDistance2[u] = checkDistance[u];
+                checkDistance[u] = d;
+            } else if (d < checkDistance2[u])
+                checkDistance2[u] = d;
+        });
+    }
+
+    const auto &curDistance = parallel ? distanceGlobal[omp_get_thread_num()] : distance;
+    const auto &curDistance2 = parallel ? distance2Global[omp_get_thread_num()] : distance2;
+    G->forNodes([&](node u) {
+        if (group.find(u) != group.end())
+            return;
+        assert(std::abs(curDistance2[u] - checkDistance2[u]) <= 1e-9);
+        assert(std::abs(curDistance[u] - checkDistance[u]) <= 1e-9);
+    });
+}
+#endif
+
+template class GroupClosenessLocalSearchImpl<count>;
+template class GroupClosenessLocalSearchImpl<edgeweight>;
+} // namespace
+
+using GroupClosenessLocalSearchUnweighted = GroupClosenessLocalSearchImpl<count>;
+using GroupClosenessLocalSearchWeighted = GroupClosenessLocalSearchImpl<edgeweight>;
+
+GroupClosenessLocalSearch::GroupClosenessLocalSearch(const Graph &G, const std::vector<node> &group,
+                                                     bool runGrowShrink, count maxIterations)
+    : weighted(G.isWeighted()) {
+
+    if (weighted)
+        impl = std::make_unique<GroupClosenessLocalSearchWeighted>(G, group, runGrowShrink,
+                                                                   maxIterations);
+    else
+        impl = std::make_unique<GroupClosenessLocalSearchUnweighted>(G, group, runGrowShrink,
+                                                                     maxIterations);
+}
+
+// We need to call the correct destructor
+GroupClosenessLocalSearch::~GroupClosenessLocalSearch() {
+    if (weighted)
+        delete static_cast<GroupClosenessLocalSearchWeighted *>(impl.release());
+    else
+        delete static_cast<GroupClosenessLocalSearchUnweighted *>(impl.release());
```

Comment:
This shouldn't be necessary. The unique pointer has type `Algorithm`, which has a virtual destructor and thus the destructor of the inherited classes that also have virtual descructor should be called.